From b31916fbb015e03a009995c2d9865ceb31627056 Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Tue, 12 Mar 2024 12:59:45 +0100
Subject: [PATCH 1086/1141] arm64: dts: st: Add lpuart1 in the stm32mp257f-ev1
 board

Add lpuart1 node in stm32mp257f-ev1.dts and in the serial aliases.

Change-Id: I08bd76f1d1ea34bb7bd9b14e2a57937f0715d9ce
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/366064
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -25,6 +25,7 @@
 		ethernet1 = &eth1;
 		serial0 = &usart2;
 		serial1 = &usart6;
+		serial2 = &lpuart1;
 	};
 
 	chosen {
@@ -425,6 +426,18 @@
 	status = "okay";
 };
 
+&lpuart1 {
+	pinctrl-names = "default", "idle", "sleep";
+	pinctrl-0 = <&lpuart1_pins_a>;
+	pinctrl-1 = <&lpuart1_idle_pins_a>;
+	pinctrl-2 = <&lpuart1_sleep_pins_a>;
+	/delete-property/dmas;
+	/delete-property/dma-names;
+	wakeup-source;
+	uart-has-rtscts;
+	status = "disabled";
+};
+
 &ltdc {
 	default-on;
 	rotation-memory = <&ltdc_sec_rotation>;
